File: Golang/Easy/20_Valid_Parentheses/solution.go
package validparentheses

func isValid(s string) bool {
	brackets := []string{}
	bracketMap := map[string]string{
		")": "(",
		"]": "[",
		"}": "{",
	}

	for _, v := range s {
		c := string(v)
		lastIndex := len(brackets) - 1
		if v, exist := bracketMap[c]; exist {
			if lastIndex < 0 {
				return false
			}
			check := brackets[lastIndex]
			brackets = brackets[:lastIndex]
			if check != v {
				return false
			}
		} else {
			brackets = append(brackets, c)
		}
	}

	if len(brackets) != 0 {
		return false
	}
	return true
}

File: Python/Easy/20_Valid_Parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        brackets = []
        for c in s:
            if c == ")":
                check = brackets.pop()
                if check != "(":
                    return False
            elif c == "]":
                check = brackets.pop()
                if check != "[":
                    return False
            elif c == "}":
                check = brackets.pop()
                if check != "{":
                    return False
            else:
                brackets.append(c)
        if len(brackets) != 0:
            return False
        return True
